packages/features/flags/features.repository.ts

```diff
@@ -70,62 +121,83 @@ export class FeaturesRepository implements IFeaturesRepository {
-        select: { id: true },
-      });
-      if (user) return true;
-      return false;
+      const query = Prisma.sql`
+        WITH RECURSIVE TeamHierarchy AS (
+          -- Start with teams the user belongs to
+          SELECT t.id, t."parentId"
+          FROM "Team" t
+          INNER JOIN "Membership" m ON m."teamId" = t.id
+          WHERE m."userId" = ${userId}
+
+          UNION ALL
+
+          -- Recursively get parent teams
+          SELECT t.id, t."parentId"
+          FROM "Team" t
+          INNER JOIN TeamHierarchy th ON t.id = th."parentId"
+        )
+        -- Check if any team in the hierarchy has the feature
+        SELECT 1
+        FROM "TeamFeatures" tf
+        WHERE tf."featureId" = ${slug}
+        AND tf."teamId" IN (SELECT id FROM TeamHierarchy)
+        LIMIT 1;
+      `;
+
+      const result = await db.$queryRaw<unknown[]>(query);
+      return result.length > 0;
     } catch (err) {
       captureException(err);
       throw err;
     }
   }
 
   /**
-   * Checks if a team has access to a specific feature.
-   * Also checks if the team's parent organization has the feature.
-   * @param teamId - The ID of the team to check
+   * Checks if a team or any of its ancestors has access to a specific feature.
+   * Uses a recursive CTE raw SQL query for performance.
+   * @param teamId - The ID of the team to start the check from
    * @param featureId - The feature identifier to check
-   * @returns Promise<boolean> - True if the team has access to the feature, false otherwise
-   * @throws Error if the team feature check fails
+   * @returns Promise<boolean> - True if the team or any ancestor has the feature, false otherwise
+   * @throws Error if the database query fails
    */
-  async checkIfTeamHasFeature(teamId: number, featureId: keyof AppFlags) {
+  async checkIfTeamHasFeature(teamId: number, featureId: keyof AppFlags): Promise<boolean> {
     try {
-      const teamFeature = await db.teamFeatures.findFirst({
-        where: {
-          OR: [
-            { teamId, featureId },
-            {
-              team: {
-                parent: {
-                  features: {
-                    some: {
-                      featureId,
-                    },
-                  },
-                },
-              },
-            },
-          ],
-        },
-      });
-      return !!teamFeature;
+      // NOTE: Ensure table ("Team", "TeamFeatures") and column ("id", "parentId", "featureId", "teamId")
+      // names exactly match your Prisma schema.
+      const query = Prisma.sql`
+        WITH RECURSIVE Ancestors AS (
+          -- Anchor member: Start with the initial team
+          SELECT id, "parentId"
+          FROM "Team"
+          WHERE id = ${teamId}
+
+          UNION ALL
+
+          -- Recursive member: Find the parent of the team found in the previous step
+          SELECT T.id, T."parentId"
+          FROM "Team" T
+          INNER JOIN Ancestors A ON T.id = A."parentId"
```

Comment:
I think we can early return and stop the recursion if any of the teams have a feature
```suggestion
  -- Recursive member: Find the parent of the team found in the previous step
  SELECT T.id, T."parentId"
  FROM "Team" T
  INNER JOIN Ancestors A ON T.id = A."parentId"
  WHERE T."parentId" IS NOT NULL
  -- Stop recursion if we find the feature in any ancestor
  AND NOT EXISTS (
    SELECT 1 
    FROM "TeamFeatures" TF 
    WHERE TF."teamId" = T.id 
    AND TF."featureId" = ${featureId}
  )
```

Reply:
Havent tested - just food for thought